crates/markdown_preview/src/markdown_elements.rs

```diff
@@ -35,6 +41,13 @@ impl ParsedMarkdownElement {
     }
 }
 
+#[derive(Debug)]
+#[cfg_attr(test, derive(PartialEq))]
+pub enum MarkdownParagraph {
```

Comment:
Can we change this to "MarkdownParagraphChunk"?

And the names of these types to just "Text" and "Image"? Perhaps with a `type MarkdownParagraph = Vec<MarkdownParagraphChunk>`? This way it's clear why markdown paragraphs are represented as a Vec<Vec<MarkdownParagraphChunk>>